src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,142 @@
+                }
+                tasks.add(t);
+                break;
+            case "D":
+                Deadline d = new Deadline(line[2],
+                        LocalDateTime.parse(line[3], DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm")));
+                if (line[1].equals("1")) {
+                    d.markAsDone();
+                }
+                tasks.add(d);
+                break;
+            case "E":
+                Event e = new Event(line[2],
+                        LocalDateTime.parse(line[3], DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm")),
+                        LocalDateTime.parse(line[4], DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm")));
+                if (line[1].equals("1")) {
+                    e.markAsDone();
+                }
+                tasks.add(e);
+                break;
+            }
+        }
+    }
+
+    /**
+     * Constructor for empty tasklist.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Deletes specified task from memory.
+     * @param index Task number to be deleted
+     * @throws DukeException errors that occur due to invalid user input
+     */
+    public void delete(int index) throws DukeException {
+        if (index < 0 || index > tasks.size()-1) {
+            throw new DukeException("Wow, deleting a nonexistent task? Check your tasks again with 'list'.");
+        }
+        Task t = tasks.get(index);
+        tasks.remove(t);
+        String returnLine = "Looks like you have more time to sleep now. Deleted this for you:\n"
+                + t.toString()
+                + "\nYou now have " + tasks.size() + " things to do.\n";
+        System.out.println(returnLine);
+    }
+
+    /**
+     * Adds a new Deadline task.
+     * @param task Deadline to be added
+     */
+    public void addDeadline(Deadline task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Adds a new Event task.
+     * @param task Event to be added
+     */
+    public void addEvent(Event task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Adds a new ToDo task.
+     * @param task ToDo to be added
+     */
+    public void addToDo(ToDo task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Adds a new Task into the tasklist.
+     * @param task Task to be added
+     */
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Retrieves specified task from tasklist
+     * @param index task number to be retrieved
+     * @return task specified
+     */
+    public Task getTask(int index) {
+        return tasks.get(index);
+    }
+
+    /**
+     * Returns number of tasks in tasklist
+     * @return number of tasks in tasklist
+     */
+    public int getLength() {
+        return tasks.size();
+    }
+
+    /**
+     * Returns a string representation of the tasks in the tasklist.
+     * @return Tasks to write into save file
+     */
+    public String[] toWriteString() {
```

Comment:
I noticed that you used the same name 'toWriteString' for various methods. Some return a String, others return a String array.
Is there a way to differentiate these method names so that they won't be as confusing?

Also, 'toWriteString' doesn't quite sound like a verb. I presume it's based on 'toString', but is there a way to name these functions so they are more descriptive?